### ISSUE TITLE
Steam games not opening

### ISSUE THREAD

**Issue by colinmarc** [colinmarc/magic-mirror#6]
From @purekillma (just moving over from #5 so it's easier for me to keep track of):

> Right now messing around with steam and noticing quite a few games don't launch, but launch fine not through magic mirror. Not sure if this is related to them opening up in windowed mode or what. Also, if I run steam in non big picture and try to open properties of a game, it crashes. Again not sure if it is because it is opening another window.
> Update, it seems like any game that doesn't have native linux support ie: running with proton, doesn't work.

https://github.com/colinmarc/magic-mirror/assets/83978127/f3550cf3-598a-48dc-b37a-e7eebb968ac6

[mmserver.log](https://github.com/colinmarc/magic-mirror/files/15179175/mmserver.log)
[steam-34130.log](https://github.com/colinmarc/magic-mirror/files/15179176/steam-34130.log)

@purekillma, can you try a few things for me? Thank you so much for bearing with me, and sorry that you're hitting so many bugs right off the bat.

 1. Try changing `command` in the config to `["steam", "-steamos", "-gamepadui"]` (this launches big-picture mode). I think things work a bit better in that mode.
 2. Try adding another app called `glxinfo` with the following contents: `command = ["sh", "-c", "glxinfo; eglinfo"]`, running it (it should exit immediately), and uploading the **client** log from the bug report (called `glxinfo-XXXX.log`)

I will add step 2. to happen automatically with `--bug-report`, for the next version that I release of MM.

**Comment by purekillma** [comment]
I get the same results in big picture mode as well, do you want glxinfo to be ran "xwayland = true" as well? If so, here is the bug report for glxinfo
[sh-85684.log](https://github.com/colinmarc/magic-mirror/files/15188306/sh-85684.log)

**Comment by colinmarc** [comment]
Hm, this is probably the issue! On my machine, I get this (I'm using a 6650 XT):

```
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: AMD (0x1002)
    Device: AMD Radeon RX 6650 XT (radeonsi, navi23, LLVM 16.0.6, DRM 3.57, 6.8.5-1-MANJARO) (0x73ef)
    Version: 24.1.0
    Accelerated: yes
    Video memory: 8192MB
```

But yours is:

```
OpenGL vendor string: Mesa
OpenGL renderer string: llvmpipe (LLVM 17.0.6, 256 bits)
OpenGL core profile version string: 4.5 (Core Profile) Mesa 24.2.0-devel (git-0bafa94f9a)
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
```

So for some reason Steam and games aren't launching with the GPU - they're using the CPU for rendering, which is probably causing the bug.

Did you install mesa from mesa-git or build it yourself?

**Comment by purekillma** [comment]
That would make sense, since I am not super familiar with Arch based distros, I switched back to Ubuntu 24.04 and installed a managed PPA - ppa:ernstp/mesaaco which has mesa 24.1 in it. Upon doing this and setting up your application and running glxinfo, it is now reporting as my GPU as the renderer! Also, this fixed my issue if steam starting slowly and timing out, which I believe you referenced elsewhere could be a potential issue with CPU usage when not using hardware encoding.

I tested a few games that didn't work and now they seem to be working! I think things will get a lot easier when 24.1 is mainstream stable.

Sorry for bothering you so much about this, but now my next question is in regards to first person games where you use the mouse to look. Are you going to implement a feature where you can lock the mouse? Also, do controllers work?

**Comment by colinmarc** [comment]
I'm so glad it's working now! I'm working on making it a louder warning if there's no gpu acceleration available via XWayland, and surfacing more of the diagnostic information.

>  I think things will get a lot easier when 24.1 is mainstream stable.

Yeah, I thought that the needed features were already available in package managers - I maybe jumped the gun announcing the project. Sorry about that. IIUC, 24.1 should land sometime this month.

> Are you going to implement a feature where you can lock the mouse? 

This was already implemented, but I just noticed there was a bug: 003fe97034cbbd71a8845841cf9d26e592c27696. That fix will be in the next release (probably this weekend).

>  Also, do controllers work?

Unfortunately, controller support is still a ways off. It's a priority, though.

**Comment by colinmarc** [comment]
At least some of the XWayland issues should be fixed in v0.2.0 - see https://github.com/colinmarc/magic-mirror/commit/54c9724a476d023547fb1c2ccc5d74bc6eadc6a3. I'm going to close this for now.
